ios/NativeSigner/Screens/MainScreenContainer.swift

```diff
@@ -24,74 +24,74 @@ struct MainScreenContainer: View {
-                            checkSeedCollision: {seedName in return data.checkSeedCollision(seedName: seedName)},
-                            restoreSeed: {seedName, seedPhrase, createRoots in data.restoreSeed(seedName: seedName, seedPhrase: seedPhrase, createRoots: createRoots)},
-                            sign: {seedName, comment in data.sign(seedName: seedName, comment: comment)},
-                            doWipe: data.wipe,
-                            alertShow: {data.alertShow = true},
-                            increment: { seedName, details in
-                                let seedPhrase = data.getSeed(seedName: seedName)
-                                if seedPhrase != "" {
-                                    data.pushButton(action: .increment, details: "1", seedPhrase: seedPhrase)
-                                }
-                            }
+                            getSeedForBackup: {seedName in return data.getSeed(seedName: seedName, backup: true)},
+                            sign: {seedName, comment in data.sign(seedName: seedName, comment: comment)}
+                        )
+                        AlertSelector(
+                            alertData: data.actionResult.alertData, canaryDead: data.canaryDead, resetAlert: data.resetAlert, pushButton: {action, details, seedPhrase in data.pushButton(action: action, details: details, seedPhrase: seedPhrase)}
                         )
-                        Spacer()
                     }
-                    ModalSelector(
-                        modalData: data.actionResult.modalData,
-                        alert: data.alert,
-                        alertShow: {data.alertShow = true},
-                        pushButton: {action, details, seedPhrase in data.pushButton(action: action, details: details, seedPhrase: seedPhrase)},
-                        removeSeed: { seedName in data.removeSeed(seedName: seedName)},
-                        restoreSeed: {seedName, seedPhrase, createSeedKeys in data.restoreSeed(seedName: seedName, seedPhrase: seedPhrase, createRoots: createSeedKeys)},
-                        createAddress: {path, seedName in data.createAddress(path: path, seedName: seedName)},
-                        getSeedForBackup: {seedName in return data.getSeed(seedName: seedName, backup: true)},
-                        sign: {seedName, comment in data.sign(seedName: seedName, comment: comment)}
-                    )
-                    AlertSelector(
-                        alertData: data.actionResult.alertData
+                    .gesture(
+                        DragGesture().updating($dragOffset, body: { (value, state, transaction) in
+                            if value.startLocation.x < 20 && value.translation.width > 100 {
+                                data.pushButton(action: .goBack)
+                            }
+                        })
                     )
```

Comment:
Just curious, any reason not to pass `.environmentObject(data)` to `AlertSelector`?